pkg/jobs/hive/hive.go

```diff
@@ -181,3 +187,189 @@ func monitorDeployStatus(client clientv1.Client, hiveset hiveclient.Interface, c
+			}
+		}
+	}
+	if !isValidVersion {
+		return errors.New("Provided version is not valid")
+	}
+
+	klog.V(2).Info("Create managedclusterview " + clusterName)
+
+	managedclusterview := &managedclusterviewv1beta1.ManagedClusterView{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      clusterName,
+			Namespace: clusterName,
+		},
+		Spec: managedclusterviewv1beta1.ViewSpec{
+			Scope: managedclusterviewv1beta1.ViewScope{
+				Group:     "config.openshift.io",
+				Kind:      "ClusterVersion",
+				Name:      "version",
+				Namespace: "",
+				Version:   "v1",
+			},
+		},
+	}
+
+	if err := client.Create(context.TODO(), managedclusterview); err != nil {
+		return err
+	}
+	resultmcview := managedclusterviewv1beta1.ManagedClusterView{}
+	for i := 1; i <= 5; i++ {
+		time.Sleep(utils.PauseFiveSeconds)
+		if err := client.Get(context.TODO(), types.NamespacedName{
+			Namespace: clusterName,
+			Name:      clusterName,
+		}, &resultmcview); err != nil {
+			return err
+		}
+		if resultmcview.Status.Result.Raw != nil {
+			break
+		}
+	}
+
+	resultClusterVersion := resultmcview.Status.Result
+
+	clusterVersion := map[string]interface{}{}
+
+	if resultClusterVersion.Raw != nil {
+		err := json.Unmarshal(resultClusterVersion.Raw, &clusterVersion)
+		utils.CheckError(err)
+	}
+
+	var desiredVersion interface{}
+	if status, ok := clusterVersion["status"]; ok {
+		cvstatus := status.(map[string]interface{})
+		if availableUpdates, ok := cvstatus["availableUpdates"]; ok {
+			if cvAvailableUpdates, ok := availableUpdates.([]interface{}); ok {
+				for _, version := range cvAvailableUpdates {
+					if version.(map[string]interface{})["version"] == desiredUpdate {
+						desiredVersion = version
+						break
+					}
+				}
+			}
+		}
+	}
+
+	clusterVersion["spec"].(map[string]interface{})["desiredUpdate"] = desiredVersion
+	if curator.Spec.Upgrade.Channel != "" {
+		clusterVersion["spec"].(map[string]interface{})["channel"] = curator.Spec.Upgrade.Channel
+	}
+
+	if curator.Spec.Upgrade.Upstream != "" {
+		clusterVersion["spec"].(map[string]interface{})["upstream"] = curator.Spec.Upgrade.Upstream
+	}
+
+	var updateClusterVersion runtime.RawExtension
+	if clusterVersion != nil {
+		b, err := json.Marshal(clusterVersion)
+		utils.CheckError(err)
+		updateClusterVersion.Raw = b
+	}
+	klog.V(2).Info("Create managedclusteraction to update clusterversion " + clusterName)
+	managedclusteraction := &managedclusteractionv1beta1.ManagedClusterAction{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      clusterName,
+			Namespace: clusterName,
+		},
+		Spec: managedclusteractionv1beta1.ActionSpec{
+			ActionType: managedclusteractionv1beta1.UpdateActionType,
+			KubeWork: &managedclusteractionv1beta1.KubeWorkSpec{
+				Resource:       "clusterversions",
+				Name:           "version",
+				Namespace:      "",
+				ObjectTemplate: updateClusterVersion,
+			},
+		},
+	}
+	if err := client.Create(context.TODO(), managedclusteraction); err != nil {
+		return err
+	}
+	return nil
```

Comment:
do we want to delete the managedclusteraction & managedclusterview?

Reply:
MCA is deleted automatically. once its done updating. II am deleting MCV after monitor-upgrade

Reply:
yea, and do we need any status check of MCA incase it failed?

Reply:
agree.. I can add status check for MCA